### PR TITLE
Add text about Capabilities and features that can only in mode 'off' (issue: #296)

### PR DESCRIPTION
**Pull request by adam-be**

### CODE REVIEW

**Comment by adam-be** [comment]
Travis seems to be in a bad mood today. The line wrap test succeeds locally for me.

File: getusermedia.html [hunk 1]
```diff
@@ -3569,7 +3569,9 @@
       constrainable properties and values that the User Agent supports. Note
       that a User Agent MAY support only a subset of the properties defined in
       the Web platform, and MAY support a subset of the set values for those
-      properties that it does support. Note that Capabilities are returned from
+      properties that it does support. A constrainable property, representing a
+      particular feature, is considered supported even if the only allowed
```
Comment (martinthomson) [inline getusermedia.html]: I think that you need to be much clearer here.  

> An implementation might implement a particular constrainable property for one type of source and not another.  On sources where the property cannot be used, a constraint MUST be supported, but the implementation MAY limit that support to a value where the feature is disabled, or even a value that cannot be expressed.  For instance, a `width` constraint does not apply to audio sources, but this would be reported as being supported for audio sources.  Attempting to set a `width` constraint on an audio source would result in the constraint not being met.

Nit: please use paragraphs for new concepts.   This document is very hard to read as it is.

Comment (jan-ivar) [inline getusermedia.html]: I'm not sure what we're trying to say here. In my mind, a `width` property is completely orthogonal to audio sources, so audio sources will do best to ignore it completely IMHO. I didn't think the issue here was about diversity of source types, but rather about whether it would be ok for UAs to acknowledge constraints it had dud implementations of, something I think is ill advised, as I mention in https://github.com/w3c/mediacapture-main/issues/296#issuecomment-173726112.

Comment (martinthomson) [inline getusermedia.html]: This is about dealing with [getSupportedConstraints()](https://w3c.github.io/mediacapture-main/#widl-MediaDevices-getSupportedConstraints-MediaTrackSupportedConstraints).

Since the value there is global and not separated by media device type, this rule is required.

Comment (jan-ivar) [inline getusermedia.html]: Why does the fact that `getSupportedConstraints()` has a shared namespace mean that audio sources have to respond one way or another to `width`? - I don't consider `getSupportedConstraints()` a discovery mechanism. People already need to know what `width` is, what its syntax is, and what types it is meant for.

Comment (martinthomson) [inline getusermedia.html]: Right, my mistake.  So change the width example to something less obvious, like AEC.  An audio device that doesn't support echoCancellation will support only a value of false for that constraint.

Comment (jan-ivar) [inline getusermedia.html]: Right. I think what's confusing in this text is it talks about what "the User Agent supports", which is misleading, and makes it sound like we're talking about `getSupportedConstraints()`. We're not. This is about the Capabilities dictionary used in `track.getCapabilities()`, so this text should be all about what a specific *source* supports. That is the proper context that's not coming through clearly enough here IMHO.

So I think what might need clarification here is the word "supported" and what it relates to. We seem to shift between what the UA "supports" and what a source "supports".

One solution might be to avoid the word "support" entirely here, and just talk about what capabilities (or lack of capabilities) a device (aka source) has.

Comment (jan-ivar) [inline getusermedia.html]: The point in the text that the content of the dictionary may vary from browser to browser, is well taken, I just find it a bit confusing that there's no mention of the fact that the real purpose of this dictionary is to glean what individual sources can and can't do.

Comment (adam-be) [inline getusermedia.html]: Yes, the word "supported" in this text is not the same the "Supported" in ```getSupportedConstraints()``` (which perhaps should have been ```getKnownConstraintNames()```.

I agree that this text would benefit from talking more about what the specific source, that's backing up the track, supports. Someone willing to take a stab at it?

Comment (jan-ivar) [inline getusermedia.html]: I actually think `getSupportedConstraints` was named right, because IMHO it would be a mistake to return `true` for a constraint the browser didn't fully implement. `false` is the most informative to the web developer here, pointing (correctly) to the browser as the limitation, rather than any of their hardware devices (which is how I interpret a constraint failure).

Comment (adam-be) [inline getusermedia.html]: I find it a bit strange that the browser could detect that the hardware is capable enough, but the browser itself doesn't have the means to control the hardware features. Since it's the browser that's the messenger, I would assume that information about such hardware features could never reach the script.

Comment (jan-ivar) [inline getusermedia.html]: @adam-be I'm not following, could you rephrase? Are you saying there are no microphones with hardware aec that cannot be turned off, or something else?

Comment (adam-be) [inline getusermedia.html]: @jan-ivar, no. I was referring to your comment above about the ```getSupportedContraints``` method returning ```false``` when the browser is the limitation. Can you give an example when that would occur?

Comment (jan-ivar) [inline getusermedia.html]: Sorry, I meant `undefined`.

**Comment by adam-be** [comment]
This one has been overtaken by #315
